includes/classes/Feature/Facets/Facets.php

```diff
@@ -390,19 +399,102 @@ public function get_aggs( $response, $query, $query_args, $query_object ) {
 						continue;
 					}
 
-					$GLOBALS['ep_facet_aggs'][ $key ] = [];
+					$processed_aggs[ $key ] = [];
 
 					if ( ! empty( $agg['value'] ) ) {
-						$GLOBALS['ep_facet_aggs'][ $key ] = $agg['value'];
+						$processed_aggs[ $key ] = $agg['value'];
 						continue;
 					}
 
 					foreach ( $agg['buckets'] as $bucket ) {
-						$GLOBALS['ep_facet_aggs'][ $key ][ $bucket['key'] ] = $bucket['doc_count'];
+						$processed_aggs[ $key ][ $bucket['key'] ] = $bucket['doc_count'];
 					}
 				}
 			}
+
+			$this->set_query_aggregations( $query_object, $processed_aggs );
+		}
+	}
+
+	/**
+	 * Set aggregation data for a query.
+	 *
+	 * @since 5.3.0
+	 * @param \WP_Query $query        The WP_Query object to store data for
+	 * @param array     $aggregations The aggregation data to store
+	 */
+	public function set_query_aggregations( $query, $aggregations ): void {
+		// Store aggregations on the query object
+		$query->ep_aggregations = $aggregations;
+
+		if ( $query->is_main_query() && $this->should_sync_to_global( $query ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html__( 'The global variable $GLOBALS[\'ep_facet_aggs\'] is deprecated. Access aggregation data directly from the query object using $query->ep_aggregations or the Facets feature methods get_query_aggregations() and get_facet_aggregation().', 'elasticpress' ),
+				'5.3.0'
```

Comment:
```suggestion
				'ElasticPress 5.3.0'
```
Prepending ElasticPress to the version argument will make it clearer what exactly is doing something wrong.